### PR TITLE
cmd: add `dump` command to dump metadata to a file and `load` to load from a file

### LINKED ISSUE #50: Backup tool for meta
**What would you like to be added**:

Provide a tool to dump metadata as JSON format, then we can have other tool to assemble them and get data back.

**Why is this needed**:

In worsest case, if we lose Redis, we'd like to have a tool to get most of data back from S3.

### PR DESCRIPTION
**Pull request by Sandy4999**
close #50

Todo:
- [x] calculate slice refs when loading
- [ ] customize json marshal to stream encode/decode; and to make the dumped file more compact

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -81,6 +81,7 @@ func main() {
 			profileFlags(),
 			statusFlags(),
 			warmupFlags(),
+			backupFlags(),
```
Comment (davies) [inline cmd/main.go:84]: We could have dump/load as separate command

Comment (Sandy4999) [reply cmd/main.go]: c139306

File: cmd/dump.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	}
+	defer fp.Close()
+	return m.DumpMeta(fp)
+}
+
+func dumpFlags() *cli.Command {
+	return &cli.Command{
+		Name:      "dump",
+		Usage:     "dump JuiceFS metadata into a standalone file",
+		ArgsUsage: "META-ADDR [FILE (default: juicefs-metadata.dump)]",
```
Comment (davies) [inline cmd/dump.go:48]: no default value for FILE

File: pkg/meta/types.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	Uid       uint32 `json:"uid"`
+	Gid       uint32 `json:"gid"`
+	Atime     int64  `json:"atime"`
+	Mtime     int64  `json:"mtime"`
+	Ctime     int64  `json:"ctime"`
+	Atimensec uint32 `json:"atimensec"`
+	Mtimensec uint32 `json:"mtimensec"`
+	Ctimensec uint32 `json:"ctimensec"`
+	Nlink     uint32 `json:"nlink"`
+	Length    uint64 `json:"length,omitempty"`
```
Comment (davies) [inline pkg/meta/types.go:50]: 0 length of a file should not be omitted

File: pkg/meta/sql.go [hunk 4]
```diff
@@ -2281,3 +2283,314 @@ func (m *dbMeta) RemoveXattr(ctx Context, inode Ino, name string) syscall.Errno
 		return err
 	}))
 }
+
+func (m *dbMeta) dumpEntry(name string, inode Ino) (*DumpedEntry, error) {
+	e := &DumpedEntry{Name: name, Inode: inode}
+	return e, m.txn(func(s *xorm.Session) error {
```
Comment (davies) [inline pkg/meta/sql.go:2289]: Should we use repeatable read isolation level?

**Comment by davies** [comment]
@Sandy4999 Could you add a few unit tests for them?  loading a sample file into redis/db, verify some functions, then dump it again, and compare the  results.

File: cmd/dump.go (hunk 2 above)
Comment (Sandy4999) [reply cmd/dump.go]: b936bea

File: pkg/meta/types.go (hunk 3 above)
Comment (Sandy4999) [reply pkg/meta/types.go]: b936bea

**Review by davies — APPROVED** [review]
pending on unit test

**Pull request merged**